trade-finance-manager-api/api-tests/v1/deal-cancellation/submit-deal-cancellation.post.api-test.ts

```diff
@@ -0,0 +1,154 @@
+  bankRequestDate: new Date().valueOf(),
+  effectiveFrom: new Date().valueOf(),
+});
+
+describe('POST /v1/deals/:id/cancellation/submit', () => {
+  let testUsers: Awaited<ReturnType<typeof initialiseTestUsers>>;
+  let aPimUser: TestUser;
+
+  beforeAll(async () => {
+    testUsers = await initialiseTestUsers(app);
+    aPimUser = testUsers().withTeam(TEAM_IDS.PIM).one();
+  });
+
+  describe('when FF_TFM_DEAL_CANCELLATION_ENABLED is disabled', () => {
+    beforeEach(() => {
+      process.env.FF_TFM_DEAL_CANCELLATION_ENABLED = 'false';
+    });
+
+    afterAll(() => {
+      process.env = originalProcessEnv;
+    });
+
+    it('returns a 404 response for an authenticated user with a valid id path', async () => {
+      // Arrange
+      const url = getSubmitTfmDealCancellationUrl({ id: validId });
+
+      // Act
+      const response = await as(aPimUser).post(aValidPayload()).to(url);
+
+      // Assert
+      expect(response.status).toEqual(404);
+    });
+  });
+
+  describe('when FF_TFM_DEAL_CANCELLATION_ENABLED is enabled', () => {
+    beforeEach(() => {
+      process.env.FF_TFM_DEAL_CANCELLATION_ENABLED = 'true';
+    });
+
+    afterAll(() => {
+      process.env = originalProcessEnv;
+    });
+
+    withTeamAuthorisationTests({
+      allowedTeams: [TEAM_IDS.PIM],
+      getUserWithTeam: (team) => testUsers().withTeam(team).one(),
+      makeRequestAsUser: (user: TestUser) =>
+        as(user)
+          .post(aValidPayload())
+          .to(getSubmitTfmDealCancellationUrl({ id: validId })),
+      successStatusCode: 200,
+    });
+
+    it('returns a 401 response when user is not authenticated', async () => {
+      // Arrange
+      const url = getSubmitTfmDealCancellationUrl({ id: validId });
+
+      // Act
+      const response = await post(aValidPayload()).to(url);
+
+      // Assert
+      expect(response.status).toEqual(401);
+    });
+
+    it('returns a 400 response when the id path param is invalid', async () => {
+      // Arrange
+      const url = getSubmitTfmDealCancellationUrl({ id: 'invalid' });
+
+      // Act
+      const response = await as(aPimUser).post(aValidPayload()).to(url);
+
+      // Assert
+      expect(response.status).toEqual(400);
+    });
+
+    const invalidPayloads = [
+      {
+        description: 'the reason is not a string',
+        payload: { ...aValidPayload(), reason: 12 },
+      },
+      {
+        description: `the reason is over ${MAX_CHARACTER_COUNT} characters`,
+        payload: { ...aValidPayload(), reason: 'x'.repeat(MAX_CHARACTER_COUNT + 1) },
+      },
+      {
+        description: 'the reason is undefined',
+        payload: { ...aValidPayload(), reason: undefined },
+      },
+      {
+        description: 'the bankRequestDate is a string ',
+        payload: { ...aValidPayload(), bankRequestDate: new Date().toISOString() },
+      },
+      {
+        description: 'the bankRequestDate is undefined ',
+        payload: { ...aValidPayload(), bankRequestDate: undefined },
```

Comment:
I think in such POST validation tests, instead of undefined, it should be an empty string `''`, since this is what `req.body` would contain if nothing is provided - same on line 112 and 128,